File: 1-easy/problems/2319-check-if-matrix-is-x-matrix/solution.ts
function checkXMatrix(grid: any): boolean | number | string | any {
  """
  :type grid: List[List[int]]
  :rtype: bool
  """
  return all((i-j == 0 || i+j == (grid).length-1) == (grid[i][j] != 0) for i in Array.from({length: (grid}, (_, i) => i).length) for j in Array.from({length: (grid[0]}, (_, i) => i).length));
}

export { checkXMatrix };

File: 1-easy/problems/2319-check-if-matrix-is-x-matrix/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def checkXMatrix(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: bool
        """
        return all((i-j == 0 or i+j == len(grid)-1) == (grid[i][j] != 0) for i in range(len(grid)) for j in range(len(grid[0])))
